src/System.Diagnostics.Process/src/System/Diagnostics/Process.Unix.cs

```diff
@@ -449,12 +449,26 @@ private static FileStream OpenStream(int fd, FileAccess access)
         /// </remarks>
         private static void ParseArgumentsIntoList(string arguments, List<string> results)
         {
-            var currentArgument = new StringBuilder();
             bool inQuotes = false;
 
             // Iterate through all of the characters in the argument string.
             for (int i = 0; i < arguments.Length; i++)
             {
+                for (; i < arguments.Length && (arguments[i] == ' ' || arguments[i] == '\t'); i++) ;
+
+                if (i == arguments.Length)
+                    break;
+
+                results.Add(GetNextArgument(arguments, ref i, ref inQuotes));
+            }
+        }
+
+        private static string GetNextArgument(string arguments, ref int i, ref bool inQuotes)
```

Comment:
The only way this function will exit with "inQuotes  == true" is if it drops off the end of the command line (in which case, inQuotes is never looked at again.) So it should just be a local inside GetNextArgument() and not a ref parameter.

I know the code inside CoreRT has the same code smell - I would have cleaned that up if I'd foresawn I was going to recommend it as a template :-)